clients/youtube_producer/js/yt_auth/yt_authDirective.js

```diff
@@ -0,0 +1,65 @@
+module.exports = function ($window, $http, $interval, $log) {
+	'use strict';
+	return {
+		restrict: 'E',
+		link: function(scope, element, attr){
+			//TODO: use more angular.js methods instead of mix of native and angular stuff
+			var userSettings = {
+				userId: ''
+			};
+
+			var fetchPersonalData = (function () {
```

Comment:
What's the purpose of the IIFE here?

Why are you setting the sync userSettings in the async function?

This is unreliable. If you have some latency in the wires - nothing will work...